src/osd/scrubber/scrub_queue_if.h

```diff
@@ -0,0 +1,126 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab
+
+#pragma once
+
+#include "include/utime_fmt.h"
+#include "osd/osd_types.h"
+#include "osd/scrubber_common.h"
+
+#include "utime.h"
+
+namespace Scrub {
+class ScrubSchedListener;
+struct SchedEntry;
+struct sched_conf_t;
+
+
+/**
+ *  the interface used by the PgScrubber and by the ScrubJob (a component
+ *  of the PgScrubber) to access the scrub scheduling functionality.
+ *  Separated from the actual implementation, mostly due to cyclic dependencies.
+ */
+struct ScrubQueueOps {
+
+  // a mockable ceph_clock_now(), to allow unit-testing of the scrub scheduling
+  virtual utime_t scrub_clock_now() const = 0;
+
+  virtual void scrub_next_in_queue(utime_t loop_id) = 0;
+
+  /**
+   * let the ScrubQueue know that it should terminate the
+   * current search for a scrub candidate (the search that was initiated
+   * from the tick_without_osd_lock()).
+   * Will be called once a triggered scrub has past the point of securing
+   * replicas.
+   */
+  virtual void initiation_loop_done(utime_t loop_id) = 0;
```

Comment:
It's not obvious why PgScrubber or ScrubJob would need to know about or modify the behavior of the scrub search process.

Reply:
Ah, it's part of how the PG indicates to the scrub queue that it's finished.  See below for a better version of this question.